src/lib/y2storage/package_handler.rb

```diff
@@ -73,7 +73,22 @@ def add_packages(pkg_list)
     #
     def add_feature_packages(devicegraph)
       used_features = UsedStorageFeatures.new(devicegraph)
-      add_packages(used_features.feature_packages)
+      packages = used_features.feature_packages
+      packages = packages.delete_if { |pkg| unavailable_optional_package?(pkg) }
```

Comment:
NP: `delete_if` already modifies the caller, so there is no need to assign it again. In other words
```ruby
packages = used_features.feature_packages
packages.delete_if { |pkg| unavailable_optional_package?(pkg) }
```